toolchain/check/testdata/generic/template/unimplemented.carbon

```diff
@@ -8,20 +8,44 @@
 // TIP: To dump output, run:
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/generic/template/unimplemented.carbon
 
-// --- fail_unimplemented.carbon
+// --- fail_unimplemented_operator.carbon
 
 library "[[@TEST_NAME]]";
 
 // Check that we get a reasonable diagnostic for an unimplemented operation on a template dependent expression.
 fn F[template T:! type](x: T) -> i32 {
-  // CHECK:STDERR: fail_unimplemented.carbon:[[@LINE+4]]:10: error: cannot access member of interface `Core.Mul` in type `<dependent type>` that does not implement that interface [MissingImplInMemberAccess]
+  // CHECK:STDERR: fail_unimplemented_operator.carbon:[[@LINE+4]]:10: error: cannot access member of interface `Core.Mul` in type `<dependent type>` that does not implement that interface [MissingImplInMemberAccess]
   // CHECK:STDERR:   return x.n * 3;
   // CHECK:STDERR:          ^~~~~~~
   // CHECK:STDERR:
   return x.n * 3;
 }
 
-// CHECK:STDOUT: --- fail_unimplemented.carbon
+// --- fail_unimplemented_convert.carbon
+
+library "[[@TEST_NAME]]";
+
+fn F[template T:! type](x: T) {
+  // CHECK:STDERR: fail_unimplemented_convert.carbon:[[@LINE+8]]:3: error: member name of type `<dependent type>` in compound member access is not an instance member or an interface member [CompoundMemberAccessDoesNotUseBase]
+  // CHECK:STDERR:   var v: T = 0;
+  // CHECK:STDERR:   ^~~~~~~~
+  // CHECK:STDERR:
+  // CHECK:STDERR: fail_unimplemented_convert.carbon:[[@LINE+4]]:3: error: value of type `<dependent type>` is not callable [CallToNonCallable]
```

Comment:
I'm surprised we get two errors here -- my expectation is that we get an error value with the first and that suppresses further errors.. (The errors themselves are not very clear about what has gone wrong.)